packages/menu/src/Menu.spec.tsx

```diff
@@ -1,271 +1,189 @@
+  fireEvent,
   render,
   waitFor,
   waitForElementToBeRemoved,
+  within,
 } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 
 import { MenuProps } from './Menu';
 import { Menu, MenuItem, MenuSeparator } from '.';
 
 const menuTestId = 'menu-test-id';
-const trigger = <button data-testid="menu-trigger">trigger</button>;
+const menuTriggerTestId = 'menu-trigger';
+const defaultTrigger = <button data-testid={menuTriggerTestId}>trigger</button>;
 
 function waitForTimeout(timeout = 500) {
   return new Promise(res => setTimeout(res, timeout));
 }
 
-function renderMenu(props: Omit<MenuProps, 'children'> = {}) {
-  const utils = render(
+/** Renders a Menu with the given props */
+function renderMenu({
+  trigger = defaultTrigger,
+  ...rest
+}: Omit<MenuProps, 'children'> = {}) {
+  const renderResult = render(
     <>
       <div data-testid="backdrop" />
-      <Menu {...props} data-testid={menuTestId}>
+      <Menu trigger={trigger} {...rest} data-testid={menuTestId}>
         <MenuItem data-testid="menu-item-a">Item A</MenuItem>
         <MenuSeparator />
         <MenuItem href="http://mongodb.design">Item B</MenuItem>
+        <MenuItem href="http://mongodb.design">Item C</MenuItem>
       </Menu>
     </>,
   );
 
-  const backdrop = utils.getByTestId('backdrop');
-  return { ...utils, backdrop };
-}
-
-describe('packages/menu', () => {
-  test.todo('trigger renders as a function');
-  test.todo('trigger renders as a JSX element');
-  test.todo('menu appears when trigger is a function');
-  test.todo('menu appears when trigger is a JSX element');
-
-  test('menu appears on DOM when the "open" prop is set', () => {
-    const { getByTestId } = renderMenu({ open: true });
-    const menu = getByTestId(menuTestId);
-    expect(menu).toBeInTheDocument();
-  });
+  const backdropEl = renderResult.getByTestId('backdrop');
+  const triggerEl = renderResult.getByTestId(menuTriggerTestId);
+
+  /**
+   * Since menu elements won't exist until component is interacted with,
+   * call this after opening the menu.
+   * @returns Object of menu elements
+   */
+  async function findMenuElements(): Promise<{
+    menuEl: HTMLElement | null;
+    menuItemElements: Array<HTMLElement | null>;
+  }> {
+    const menuEl = await renderResult.findByTestId(menuTestId);
+    const menuItemElements = await within(menuEl).findAllByRole('menuitem');
+
+    return {
+      menuEl,
+      menuItemElements,
+    };
+  }
 
-  test('renders children to the DOM', () => {
-    const { getByText } = renderMenu({ open: true });
-    const menuItem = getByText('Item A');
-    expect(menuItem).toBeInTheDocument();
-  });
+  /** Opens the menu, and manually fires transition events */
+  async function openMenu() {
+    userEvent.click(triggerEl);
+    const menuElements = await findMenuElements();
+    fireEvent.transitionEnd(menuElements.menuEl as Element); // JSDOM does not automatically fire these events
```

Comment:
Can you remind me again what this line does?

Reply:
It manually fires the [transitionend](https://developer.mozilla.org/en-US/docs/Web/API/Element/transitionend_event) event, which is needed for `onEntered` (etc.) callbacks to fire